### PR TITLE
Fix Black Screen Transition Issue by Reusing didScreenTransitionEnd in MoneyRequestParticipantsSelector

### PR DESCRIPTION
**Pull request by brunovjk**
### Details
Reuse `didScreenTransitionEnd` to address the screen transition skeleton issue

### Fixed Issues
$ [31779](https://github.com/Expensify/App/issues/31779)
PROPOSAL: https://github.com/Expensify/App/issues/31779#issuecomment-1830039239

### Tests
1. Use a 'High Traffic' account, or any account with a large contact list.
2. Access the FAB and select "Request money."
3. Add an amount and click "Next" to initiate the transition.
4. Observe the screen transition from `.../manual` to the `../participants`.
5. Verify that the black screen issue no longer occurs during the transition.
6. Confirm that no errors are present during the screen transition skeleton.

- [x] Verify that no errors appear in the JS console

### Offline tests
Same as Tests.

- [x] Verify that no errors appear in the JS console

### QA Steps
Same as Tests.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/95647348/010e35b8-53c0-4d45-8164-44ef72c56e2e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/95647348/90e9ee98-7d78-45d3-be02-445705e9159a

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/95647348/509aecee-4daf-4f26-ac1c-f4912d083cce

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/95647348/9707007f-ebbd-427b-b892-59a7850594fd

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/95647348/8b537db3-ca2e-45e8-b160-8d1cd42074eb

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/95647348/982ed5ba-825c-4b5b-9883-5096affff25c

</details>

### CODE REVIEW

**Comment by brunovjk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by grgia** [comment]
@ntdiary assigned for review

**Comment by situchan** [comment]
@brunovjk your branch is very behind. The 2 components you made changes on are now deprecated.

**Comment by brunovjk** [comment]
> @brunovjk your branch is very behind. The 2 components you made changes on are now deprecated.

Let me fix it. One moment.

**Comment by brunovjk** [comment]
> @brunovjk your branch is very behind. The 2 components you made changes on are now deprecated.

@situchan we are using another [money request creation flow](https://github.com/Expensify/App/pull/28618) now, but I tested it and the [issue](https://github.com/Expensify/App/issues/31779) remains.
I believe we can use the same approach on the new components used. I'm still investigating and testing.

But what I do? Do I update my proposal or just the PR? Thanks.

**Comment by brunovjk** [comment]
> The 2 components you made changes on are now deprecated.

@situchan the [refactored money request navigation](https://github.com/Expensify/App/pull/28618) was merged but the initial [issue](https://github.com/Expensify/App/issues/31779) remains. We can apply the same [solution](https://github.com/Expensify/App/issues/31779#issuecomment-1842419123) proposed previously.

I updated this PR to apply changes at new components used. Also update the video screenshot.

File: src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js [hunk 1]
```diff
@@ -223,38 +228,40 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({
...
-            iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE,
-            true,
-        );
-        setNewChatOptions({
-            recentReports: chatOptions.recentReports,
-            personalDetails: chatOptions.personalDetails,
-            userToInvite: chatOptions.userToInvite,
-        });
-    }, [betas, reports, participants, personalDetails, translate, searchTerm, setNewChatOptions, iouType, iouRequestType]);
+        if (didScreenTransitionEnd) {
```
[the remark this answers is not in the document]
Comment (brunovjk) [reply src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js]: Thank you :D

**Comment by brunovjk** [comment]
> For native platforms, the current implementation seems to be not perfect yet

I'm having difficulty simulating native application.
But I'm working on it full time. I'll be back as soon as I have an update.

**Comment by brunovjk** [comment]
> For native platforms, the current implementation seems to be not perfect yet

I'm having difficulty emulating native application.
But I'm working on it full time. I'll be back as soon as I have an update.

File: src/pages/iou/request/step/IOURequestStepParticipants.js [hunk 2]
```diff
@@ -87,14 +87,17 @@ function IOURequestStepParticipants({
...
         >
-            <MoneyRequestParticipantsSelector
-                ref={(el) => (optionsSelectorRef.current = el)}
-                participants={participants}
-                onParticipantsAdded={addParticipant}
-                onFinish={goToNextStep}
-                iouType={iouType}
-                iouRequestType={iouRequestType}
-            />
+            {({didScreenTransitionEnd}) => (
```
Comment (grgia) [inline src/pages/iou/request/step/IOURequestStepParticipants.js:90]: Is there a reason to use this pattern instead of `&&` ? or `? :`

Comment (brunovjk) [reply src/pages/iou/request/step/IOURequestStepParticipants.js:90]: I opted for the `{({didScreenTransitionEnd}) => (` approach as it fit the existing component structure and worked seamlessly for the current implementation.
I explored alternatives like` ? :` and `&&`, but encountered issues with those approaches.

Comment (brunovjk) [reply src/pages/iou/request/step/IOURequestStepParticipants.js:90]: I opted for the `{({didScreenTransitionEnd}) => (` approach as it fit the existing component structure and worked seamlessly for the current implementation. I explored alternatives like `? :` and `&&`, but encountered issues with those approaches. If you have any suggestions or insights, I'd appreciate your feedback.

**Comment by brunovjk** [comment]
> For native platforms, the current implementation seems to be not perfect yet. 

I can reproduce it now. Working on it.

File: src/pages/iou/request/step/IOURequestStepParticipants.js (hunk 2 above)
Comment (brunovjk) [reply src/pages/iou/request/step/IOURequestStepParticipants.js:90]: Using the `{({didScreenTransitionEnd}) => (` approach allows us to pass the `didScreenTransitionEnd` variable to `MoneyRequestParticipantsSelector` regardless of its truthiness. This is particularly useful when we want to render a skeleton or initiate certain actions based on the value of `didScreenTransitionEnd`.

In contrast, using the `? :` or `&&` patterns for conditional rendering might skip rendering `MoneyRequestParticipantsSelector` when `didScreenTransitionEnd` is false, potentially leading to scenarios where we miss out on rendering the skeleton or initiating specific actions. I explored those alternatives (`? :` and `&&`), but encountered issues.

**Comment by brunovjk** [comment]
Hey @ntdiary and @grgia,

While investigating the inconsistency with `onLayout={(e)=>{` in native apps, the root cause remains elusive, but I've devised a solution.

I observed a lag tied to its natural delayed `onLayout` call during the transition process:
<details>
<summary>Getting height from onLayout callback:</summary>


https://github.com/Expensify/App/assets/95647348/e9634e24-e373-4bff-ab1c-927fb72cfb3c


</details>

By removing `onLayout={(e)=>{}` from `OptionsListSkeletonView` and adopting an alternative height calculation:
```
  constructor(props) {
    super(props);
    const screenHeight = Dimensions.get('window').height;
    const numItems = Math.ceil(screenHeight / CONST.LHN_SKELETON_VIEW_ITEM_HEIGHT);
    this.state = {
      skeletonViewItems: this.generateSkeletonViewItems(numItems),
    };
  }
```
The skeleton now loads with the parent, eliminating the delay and resolving our initial issue with skeleton visibility in native apps.

<details>
<summary>Getting height from Dimentions + constructor:</summary>


https://github.com/Expensify/App/assets/95647348/05f938fb-e297-4a79-a055-5265df3d7984

</details>

I've run tests and recorded POC videos. Keen to hear your thoughts on this approach.

**Comment by brunovjk** [comment]
> I'm not exactly sure if this will have any adverse effects.

We can set a threshold (between 0 and 1) for calculating the number of items, like:
```
const screenHeight = Dimensions.get('window').height;
const numItems = Math.ceil(screenHeight * CONST.SKELETON_VIEW_THRESHOLD_HEIGHT / CONST.LHN_SKELETON_VIEW_ITEM_HEIGHT);
```
But I understand the concern and agree to investigate further.

There's still a slight lag on iOS.
Also, I've noticed a peculiar behavior in `WorkspaceInvitePage` where the `onLayout` event is triggered, but the skeleton isn't visible. I'll try to find if there is an open issue about it.

**Comment by brunovjk** [comment]
I've made a few adjustments to the `OptionsListSkeletonView` component:

1. **Constructor Optimization:**
   - Skeleton items are now generated in the constructor for an immediate visual representation.
   - The computation for generating skeleton items remains lightweight to ensure swift instantiation.

2. **Dynamic Layout Adjustment:**
   - The `onLayout` event triggers an update to the skeleton items based on layout changes, ensuring a responsive layout.
   - Calculations for the number of skeleton items are adjusted dynamically based on the layout height.

These modifications aim to strike a balance between immediate rendering and dynamic adaptation to layout changes. The original code has been minimally modified to maintain its structure and readability.

I'll be conducting more tests, but feel free to review and provide feedback. 👍

**Comment by brunovjk** [comment]
> For native platforms, the current implementation seems to be not perfect yet. It seems that the `onLayout` event is triggered late 

After an in-depth investigation over the past days, I wasn't able to identify the root cause of the onLayout event not triggering at native. However, I've observed that a [PR](https://github.com/Expensify/App/pull/33567) to modify OptionsListSkeletonView.js has been effective in resolving the layout issue.

https://github.com/Expensify/App/assets/95647348/574b8c53-0737-4d26-8e40-ecccc17ad079

@ntdiary and @grgia considering these findings, I recommend closing our current PR as the black screen problem has been successfully mitigated with the introduction of `didScreenTransitionEnd`.

I appreciate the feedback and remain available for any clarification or further discussion.

**Comment by brunovjk** [comment]
@grgia and @ntdiary friendly bump to review.

**Comment by brunovjk** [comment]
Thank you.

**Comment by brunovjk** [comment]
thank you :)

**Comment by brunovjk** [comment]
> Eh, @brunovjk, are you suggesting to close this PR because you saw the introduction of `didScreenTransitionEnd` in another PR ?

No, I didn't know about it, do you know the PR number? I just wanted to move on with the work, it seems solid to me, but I appreciate your willingness to investigate, I agree, I'm available for anything.

**Comment by grgia** [comment]
@brunovjk could you clarify what you mean by this comment - https://github.com/Expensify/App/pull/32665#issuecomment-1872139662

It seems like you are implying you'd like to close this PR

**Comment by brunovjk** [comment]
> It seems like you are implying you'd like to close this PR

Sorry for any confusion. I'm new here, still getting the hang of it. When I said 'close,' I meant if we can incorporate `didScreenTransitionEnd` to fix the [black screen issue](https://github.com/Expensify/App/issues/31779). I can still reproduce it, and the changes proposed in this PR fix it, but let's wait for ntdiary's investigation. Looking forward to making my first contribution to Expensify!

**Comment by grgia** [comment]
@ntdiary could you take another look at this PR

**Comment by grgia** [comment]
approved/running auto tests!

**Comment by brunovjk** [comment]
Thank you very much for the investigation @ntdiary, I will update the videos in a moment.
> afraid it may be considered a regression. 

I think it's worth [controlling the skeleton](https://github.com/Expensify/App/issues/31779#issuecomment-1842806274) with an `isLoaded` state anyway,
What would be the negative side of that? :)

**Comment by brunovjk** [comment]
@ntdiary I updated all the videos with my latest changes about move 'isOptionsDataReady' to an 'useState'
If necessary, I revert and redo it, let me know please.
Thanks.

**Comment by brunovjk** [comment]
> I just found a new problem with the latest changes.

sh**, let me investigate. Do you have any ideas? Maybe revert that last commit. I'm still afraid about possible 'regression'.

**Comment by brunovjk** [comment]
Yes, true 😂 Great then, I'll try it here too. I'll come back if I find something, I'm here if you need anything.

**Comment by brunovjk** [comment]
Very good @ntdiary.
They changed `MoneyTemporaryForRefactorRequestParticipantsSelector.js`, hence the conflict. They added a `useMemo` to the const that uses `getFilteredOptions`, part of the initial problem, the black screen, was solved, but it is still not perfect, we are having the same problems in relation to the skeleton. I'm testing some methods too, I want it to be the same when you click on `FAB > StartChat`, I like the behavior of `newChatPage`, the transition runs smoothly with the skeleton. I'll keep testing here, I will try this `call the blur in BaseOptionsSelector` you said. I'm focused on making this transition perfect :D
If you have any news, I'm here. Have a good night.

**Comment by grgia** [comment]
I just want to say, great work so far on this one! @brunovjk @ntdiary 

If this PR is not finalized, could we add a [WIP] to the title? Otherwise, may we merge main for reviews? Thanks @brunovjk 😁

**Comment by brunovjk** [comment]
Thank you @grgia!!!!
I'm still testing some methods on how to control the skeleton better.
There is even an open [PR](https://github.com/Expensify/App/pull/34443), I believe they are trying to resolve a regression. However, I tested it and it generates the same problem as this one, the skeleton is still not good.

**This PR will read for review soon** :D

**Comment by brunovjk** [comment]
>  only one seems to be working (call the blur in BaseOptionsSelector)

@ntdiary, we are refactoring components that use `OptionsSelector` to switch to `SelectionList:
-  "This [issue](https://github.com/Expensify/App/issues/20354) keeps track of Phase 3 of the Selection List Refactor."
-  "Thoroughly discussed in the parent issue: #11795 [comment](https://github.com/Expensify/App/issues/11795#issuecomment-1573880715)"

But the initial [issue](https://github.com/Expensify/App/issues/31779) of this PR always come back, since in the [money request participants selector list refactoring](https://github.com/Expensify/App/pull/32692), we never use `didScreenTransitionEnds` from the parent `ScreenWrapper` to control the call for `OptionsUtils.getOptions` appropriately.

I tested the use of `didScreenTransitionEnds` in the new refactored `MoneyResquestParticipantsSelector + SelectionList` and **now** it's looking really good to me.

**Comment by brunovjk** [comment]
We also can see other **problems** caused by this refactoring:
   - `spinner loading` not show when we type a new search, instead the `skeleton list` flicks. **We have a [PR](https://github.com/Expensify/App/pull/34443) for this already, but I tested and did fix the issue, since the RCA is the missing prop `isLoadingNewOptions` in `BaseSelectionList`
   - `ReferralProgramCTA` component icon has a visual bug. A college will raise a PR, slack [conversation](https://expensify.slack.com/archives/C01GTK53T8Q/p1705068053060889), but I went ahead and adjusted the 'icon style' for the `MoneyRequestReferralProgramCTA`, we still have to search and fix for others components.
 
Those same **problems** above can be seen at the [SearchPage list refactor](https://github.com/Expensify/App/pull/33080) and [New Chat page list refactor](https://github.com/Expensify/App/pull/34356)

I fixed those **problems** by:
   - Adding back a prop `isLoadingNewOptions` in `BaseSelectionList` in order to control the `spinner loading` when we type a new search.
   - Adjust  `SelectionList` props: `sections`, `showLoadingPlaceholder` and `isLoadingNewOptions` in order to control the 'skeleton loading/section list' rendering using `didScreenTransitionEnd` and `isOptionsDataReady`.
   - Added back `fill={theme.icon}` to the `ReferralProgramCTA` component.

**Comment by brunovjk** [comment]
Committed here, changes requested by @lukemorawski here https://github.com/Expensify/App/pull/34443#issuecomment-1892308955 to resolve https://github.com/Expensify/App/pull/32692#issuecomment-1888223436.

**Comment by brunovjk** [comment]
PR Ready for review

**Comment by brunovjk** [comment]
@ntdiary PR Ready for review.

**Comment by brunovjk** [comment]
@ntdiary PR Ready for review :D

**Comment by lukemorawski** [comment]
@brunovjk looks good man. Thanks for applying my changes to your PR

**Comment by brunovjk** [comment]
Thanks @ntdiary!
> please feel free to request another C+ to take over

I dont know 😅 What do you think @lukemorawski @grgia ?

**Comment by situchan** [comment]
I can take this over if needed

**Review by ntdiary — CHANGES_REQUESTED** [review]
Thank you all for your patience. My build system has been fixed. :)
If we want to add `isLoadingNewOptions` in this PR, perhaps it would also be better to change the condition here: (not a blocker)
https://github.com/Expensify/App/blob/a8acf44090c9aab4e9af7967a79758472ada8fc3/src/components/SelectionList/BaseSelectionList.js#L434

```js
{!isLoadingNewOptions && Boolean(headerMessage) && (
```
reason:
https://github.com/Expensify/App/blob/a8acf44090c9aab4e9af7967a79758472ada8fc3/src/components/OptionsList/BaseOptionsList.js#L261-L263

Additionally, I don't think it's necessary to modify `MoneyRequestParticipantsPage.js` and `MoneyRequestParticipantsSelector.js`, as we will remove them in issue #29107.

File: src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js [hunk 3]
```diff
@@ -94,6 +101,16 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({
      */
     const [sections, newChatOptions] = useMemo(() => {
         const newSections = [];
+        if (!didScreenTransitionEnd) {
+            return [
```
[the remark this answers is not in the document]
Comment (brunovjk) [reply src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js]: done :)

**Comment by brunovjk** [comment]
> If we want to add isLoadingNewOptions in this PR, perhaps it would also be better to change the condition here

Done.

> we can simplify it to return [newSections, {}];

Done.

@ntdiary I tested it again here, and it looks like we're really good now!

**Comment by brunovjk** [comment]
The ['Selection List' migration](https://github.com/Expensify/App/pull/34008) generated some bugs in the list:
Latest 'main' version:
image 
image



This PR does not introduce nor fix these errors.

**Comment by brunovjk** [comment]
Sorry for the mess, yesterday after the [merge of a TS migration](https://github.com/Expensify/App/pull/34008) it generated some bugs and conflicts here. Today they already [reversed this migration](https://github.com/Expensify/App/pull/34733), which is why there is so much revert commit here.

@grgia and @ntdiary I just tested it and our transition continues to work great.

**Comment by brunovjk** [comment]
> fewer files may be affected

Sorry, my bad. Now Its all clean :) Thank you :D

**Comment by brunovjk** [comment]
Very good @ntdiary, good clarifications.

> 1. If you want to submit only the our origin first part in this PR, I think it's acceptable. And then put `isLoadingNewOptions`, `debouncedSearchTerm`, and the responding issue in a new PR

I think it's the best option, it was getting too much for just one PR. And as [we discussed](https://github.com/Expensify/App/issues/31779#issuecomment-1842225529), we do have other issues, but let's focus on the reuse `didScreenTransitionEnd` to address the screen transition skeleton issue.

Can I make the necessary changes and submit another commit?

**Comment by brunovjk** [comment]
> updated code again. :)

Done! I had already done it before hehe I tested and its looking good.
Thank you very much! May you have a great weekend.

**Review by ntdiary — APPROVED** [review]
LGTM. :)

File: src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js [hunk 4]
```diff
@@ -335,8 +346,9 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({
                 onSelectRow={addSingleParticipant}
                 footerContent={footerContent}
                 headerMessage={headerMessage}
-                showLoadingPlaceholder={isSearchingForReports}
+                showLoadingPlaceholder={!(didScreenTransitionEnd && isOptionsDataReady)}
                 rightHandSideComponent={itemRightSideComponent}
+                isLoadingNewOptions={isSearchingForReports}
```
[the remark this answers is not in the document]
Comment (brunovjk) [reply src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js]: Done :D

**Comment by brunovjk** [comment]
Thank you very much @ntdiary, I really appreciate your patience and goodwill :D
Now just wait for @grgia right?

**Review by lakchote — DISMISSED** [review]

**Comment by lakchote** [comment]
LGTM 👍 thanks for your contribution @brunovjk and your comments/tests @ntdiary.

**Comment by brunovjk** [comment]
I think is fixed :D
_"Jack
This is fixed now, please merge main in your PRs! [#slack_conversation](https://expensify.slack.com/archives/C01GTK53T8Q/p1706234476558379?thread_ts=1706231892.976039&cid=C01GTK53T8Q)"_

**Review by lakchote — APPROVED** [review]

**Pull request merged**